Libraries/D3D12RaytracingFallback/src/FallbackLayerUnitTests/fallbacklayerunittests.cpp

```diff
@@ -1003,6 +1045,357 @@ namespace FallbackLayerUnitTests
+                sizeof(float) * numFloats,
+                &pUpdatedVertexBuffer);
+
+            D3D12_RAYTRACING_GEOMETRY_DESC geometryDesc = {};
+            geometryDesc.Type = D3D12_RAYTRACING_GEOMETRY_TYPE_TRIANGLES;
+            geometryDesc.Triangles.IndexBuffer = 0;
+            geometryDesc.Triangles.IndexCount = 0;
+            geometryDesc.Triangles.IndexFormat = DXGI_FORMAT_UNKNOWN;
+            geometryDesc.Triangles.Transform = 0;
+            geometryDesc.Triangles.VertexFormat = DXGI_FORMAT_R32G32B32_FLOAT;
+            geometryDesc.Triangles.VertexCount = numVertices;
+            geometryDesc.Triangles.VertexBuffer.StartAddress = pVertexBuffer->GetGPUVirtualAddress();
+            geometryDesc.Triangles.VertexBuffer.StrideInBytes = floatsPerVertex * sizeof(float);
+
+            auto heapProperties = CD3DX12_HEAP_PROPERTIES(D3D12_HEAP_TYPE_DEFAULT);
+
+            D3D12_GET_RAYTRACING_ACCELERATION_STRUCTURE_PREBUILD_INFO_DESC getPrebuildDesc = {};
+            getPrebuildDesc.DescsLayout = D3D12_ELEMENTS_LAYOUT_ARRAY;
+            getPrebuildDesc.NumDescs = 1;
+            getPrebuildDesc.pGeometryDescs = &geometryDesc;
+            getPrebuildDesc.Type = D3D12_RAYTRACING_ACCELERATION_STRUCTURE_TYPE_BOTTOM_LEVEL;
+            getPrebuildDesc.Flags = buildFlags;
+
+            D3D12_RAYTRACING_ACCELERATION_STRUCTURE_PREBUILD_INFO prebuildInfo;
+            pBuilder->GetRaytracingAccelerationStructurePrebuildInfo(&getPrebuildDesc, &prebuildInfo);
+
+            CComPtr<ID3D12Resource> pBottomLevelResource;
+            D3D12_GPU_VIRTUAL_ADDRESS bottomLevelGpuVA;
+            auto accelerationStructureDesc = CD3DX12_RESOURCE_DESC::Buffer(prebuildInfo.ResultDataMaxSizeInBytes, D3D12_RESOURCE_FLAG_ALLOW_UNORDERED_ACCESS);
+            device.CreateCommittedResource(&heapProperties, D3D12_HEAP_FLAG_NONE, &accelerationStructureDesc, D3D12_RESOURCE_STATE_COMMON, nullptr, IID_PPV_ARGS(&pBottomLevelResource));
+            bottomLevelGpuVA = pBottomLevelResource->GetGPUVirtualAddress();
+
+            CComPtr<ID3D12Resource> pScratchBufferResource;
+            auto scratchBufferDesc = CD3DX12_RESOURCE_DESC::Buffer(prebuildInfo.ScratchDataSizeInBytes, D3D12_RESOURCE_FLAG_ALLOW_UNORDERED_ACCESS);
+            device.CreateCommittedResource(&heapProperties, D3D12_HEAP_FLAG_NONE, &scratchBufferDesc, D3D12_RESOURCE_STATE_COMMON, nullptr, IID_PPV_ARGS(&pScratchBufferResource));
+
+            CComPtr<ID3D12GraphicsCommandList> pCommandList;
+            m_d3d12Context.GetGraphicsCommandList(&pCommandList);
+
+            D3D12_BUILD_RAYTRACING_ACCELERATION_STRUCTURE_DESC bottomLevelDesc = {};
+            bottomLevelDesc.DestAccelerationStructureData.StartAddress = bottomLevelGpuVA;
+            bottomLevelDesc.DestAccelerationStructureData.SizeInBytes = prebuildInfo.ResultDataMaxSizeInBytes;
+            bottomLevelDesc.ScratchAccelerationStructureData.StartAddress = pScratchBufferResource->GetGPUVirtualAddress();
```

Comment:
I have a strong suspicion that the bug I pointed out with the hierarchy buffer being referenced in the ComputeAABB.hlsli is being hidden because you're re-using the scratch buffer here (which is okay from an app perspective, but I suspect it's actually only passing because it's using the hierarchy buffer in the second pass that just happens to still be lingering around from the first build). I'd recommend making 2 scratch buffers and using a clean scratch buffer for the update to confirm this